### PR TITLE
development/cargo-vendor-filterer: Fix FTB on arm.

### PR DESCRIPTION
**Pull request by fourtysixand2**
@sbo-bot: build arm development/cargo-vendor-filterer

### CODE REVIEW

**Comment by willysr** [comment]
@fourtysixand2 try again after rebased to rust16 branch I just pushed

**Comment by willysr** [comment]
that's because you are using master which hasn't got rust16 updated with new version yet
I pushed a new branch with updated rust16 and this PR is rebased on top of that branch so it will build the latest rust16 first and then your changes in cargo-vendor will be applied

**Comment by willysr** [comment]
@sbo-bot: build arm development/cargo-vendor-filterer

**Comment by willysr** [comment]
well, probably time to use manual test

**Comment by aclemons** [comment]
@sbo-bot: build arm development/cargo-vendor-filterer

**Comment by aclemons** [comment]
@sbo-bot: build arm development/cargo-vendor-filterer

**Comment by willysr** [comment]
missing kernel-modules package?

File: development/cargo-vendor-filterer/cargo-mkvendored.sh [hunk 1]
```diff
@@ -97,7 +103,7 @@ fi
 # Configure cargo-vendor-filterer
   cat << EOF >> Cargo.toml
 [package.metadata.vendor-filter]
-platforms = ["x86_64-unknown-linux-gnu", "i686-unknown-linux-gnu", "aarch64-unknown-linux-gnu"]
+platforms = ["x86_64-unknown-linux-gnu", "i686-unknown-linux-gnu", "aarch64-unknown-linux-gnu", "arm-unknown-linux-gnueabihf"]
```
Comment (aclemons) [inline development/cargo-vendor-filterer/cargo-mkvendored.sh:106]: I think this should be `armv7-unknown-linux-gnueabihf`:

```sh
docker run --rm --platform linux/armv7 aclemons/slackware:15.0-full rustc -Vv
rustc 1.58.1
binary: rustc
commit-hash: unknown
commit-date: unknown
host: armv7-unknown-linux-gnueabihf
release: 1.58.1
LLVM version: 13.0.0
```

```sh
docker run --rm --platform linux/386 aclemons/slackware:15.0-full rustc -Vv
rustc 1.58.1
binary: rustc
commit-hash: unknown
commit-date: unknown
host: i686-unknown-linux-gnu
release: 1.58.1
LLVM version: 13.0.0
```

```sh
docker run --rm --platform linux/amd64 aclemons/slackware:15.0-full rustc -Vv
rustc 1.58.1
binary: rustc
commit-hash: unknown
commit-date: unknown
host: x86_64-unknown-linux-gnu
release: 1.58.1
LLVM version: 13.0.0
```

```sh
docker run --rm --platform linux/arm64/v8 aclemons/slackware:current-full rustc -Vv
rustc 1.70.0 (90c541806 2023-05-31) (built from a source tarball)
binary: rustc
commit-hash: 90c541806f23a127002de5b4038be731ba1458ca
commit-date: 2023-05-31
host: aarch64-unknown-linux-gnu
release: 1.70.0
LLVM version: 16.0.2
```

Comment (aclemons) [reply development/cargo-vendor-filterer/cargo-mkvendored.sh:106]: Hmm, I see. The binary repack of rust is different:

```sh
rustc -Vv
rustc 1.76.0 (07dca489a 2024-02-04)
binary: rustc
commit-hash: 07dca489ac2d933c78d3c5158e3f43beefeb02ce
commit-date: 2024-02-04
host: arm-unknown-linux-gnueabihf
release: 1.76.0
LLVM version: 17.0.6
```

Comment (aclemons) [reply development/cargo-vendor-filterer/cargo-mkvendored.sh:106]: Ok, sorry for the noise. It is not this target, but the one in the slackbuild.

**Comment by aclemons** [comment]
> I expected it to fail on that, just wanted to be sure. I'll be adding an if statement.

:+1: Yes, I was misreading the error - `--target $ARCH-unknown-linux-gnu` was making it fail since it needs have the suffix `gnueabihf`. (Or no --target and it chooses automatically)

**Comment by willysr** [comment]
thanks
merged

**Pull request closed without merging**